agent/acs/handler/payload_handler.go

```diff
@@ -274,7 +286,7 @@ func (payloadHandler *payloadRequestHandler) handleUnrecognizedTask(task *ecsacs
 	}
 
 	// Only need to stop the task; it brings down the containers too.
-	eventhandler.AddTaskEvent(api.TaskStateChange{
+	payloadHandler.taskHandler.AddTaskEvent(api.TaskStateChange{
```

Comment:
@aaithal  do you mind clarifying? what exactly do you mean by "seems like leaking the abstraction"? thanks =]